test/relations.test.js

```diff
@@ -4980,6 +4980,72 @@ describe('relations', function() {
       });
     });
 
+    it('should pass options when removed by id', function(done) {
+      const verifyOptions = function(ctx, next) {
+        if (!ctx.options || !ctx.options.verify) {
+          return next(new Error('options or options.verify is missing'));
+        }
+        return next();
+      };
+      Person.observe('before save', verifyOptions);
+      Person.findOne(function(err, p) {
+        p.addressList.create({street: 'options 1'}, {verify: true}, function(err, address) {
+          if (err) {
+            Person.clearObservers('before save');
+            return done(err);
+          }
+          p.addressList.destroy(address.id, {verify: true}, function(err) {
+            if (err) {
+              Person.clearObservers('before save');
+              return done(err);
+            }
+            Person.findById(p.id, function(err, verify) {
+              if (err) {
+                Person.clearObservers('before save');
+                return done(err);
+              }
+              verify.addresses.should.have.length(1);
```

Comment:
Nitpick: from a test perspective, it would be good if you deleted all the address instances from the `Person` model before your two new tests so that here we expect none. If not, then can we leave a comment here saying there was one already?